vnext/Desktop.UnitTests/ScriptStoreTests.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <BaseScriptStoreImpl.h>
+#include <CppUnitTest.h>
+
+// Windows API
+#include <Windows.h>
+
+// Standard Library
+#include <cstdio>
+#include <cstdlib>
+
+using namespace facebook::jsi;
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+
+using std::make_shared;
+using std::make_unique;
+using std::unique_ptr;
+
+namespace Microsoft::JSI::Test {
+
+TEST_CLASS (ScriptStoreIntegrationTest) {
+  TEST_METHOD(V8_PersistPreparedScript) {
+    unique_ptr<PreparedScriptStore> preparedScriptStore = nullptr;
+
+    char tempPath[MAX_PATH];
+    if (GetTempPathA(MAX_PATH, tempPath)) {
+      preparedScriptStore = make_unique<facebook::react::BasePreparedScriptStoreImpl>(tempPath);
+    } else {
+      Assert::Fail(L"Could not get temporary folder");
+    }
+
+    constexpr size_t fileSize = 4 * 1024 * 1024;
+    constexpr size_t bufferSize = fileSize + 1; // +1 for trailing \0
+    char *buffer = static_cast<char *>(std::malloc(bufferSize * sizeof(char)));
+    int cursor = 0;
+    char id = 0;
+    while (cursor < fileSize) {
+      cursor += sprintf_s(buffer + cursor, bufferSize - cursor, "var v%019X = 'x';\n", id++);
```

Comment:
>cursor += sprintf_s(buffer + cursor, bufferSize - cursor, "var v%019X = 'x';\n", id++); [](http://example.com/codeflow?start=6&length=87)

Does this line behave properly when sprintf_s fails (and returns -1) due to insufficient space remaining in the buffer?

Also, since this code does not need to be super performant, using std::stringstream might be more readable.

Lastly, do we even need to put valid JS in the buffer? From what I can tell, this test just test the storage and retrieval functionalities of BasePreparedScriptStoreImpl, so I think we can just use a random string for simplicity.

Reply:
1. It probably doesn't. I'll address this.
2. I'll follow up offline. Not sure how to combine `std::stringstream` with string formatting.
3. Not at all, but figured that out later on.